simpleclient_common/src/test/java/io/prometheus/client/exporter/common/ExemplarTest.java

```diff
@@ -0,0 +1,391 @@
+package io.prometheus.client.exporter.common;
+
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.Counter;
+import io.prometheus.client.Gauge;
+import io.prometheus.client.Histogram;
+import io.prometheus.client.Summary;
+import io.prometheus.client.exemplars.CounterExemplarSampler;
+import io.prometheus.client.exemplars.Exemplar;
+import io.prometheus.client.exemplars.ExemplarConfig;
+import io.prometheus.client.exemplars.HistogramExemplarSampler;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.StringWriter;
+
+public class ExemplarTest {
+
+  private final String defaultTraceId = "default-trace-id";
+  private final String defaultSpanId = "default-span-id";
+  private final String customTraceId = "custom-trace-id";
+  private final String customSpanId = "custom-span-id";
+  private final String defaultExemplarLabels = "{span_id=\"" + defaultSpanId + "\",trace_id=\"" + defaultTraceId + "\"}";
+  private final String customExemplarLabels = "{span_id=\"" + customSpanId + "\",trace_id=\"" + customTraceId + "\"}";
+  private final long timestamp = System.currentTimeMillis();
+
+  private CollectorRegistry registry;
+
+  private final CounterExemplarSampler origCounterExemplarSampler = ExemplarConfig.getCounterExemplarSampler();
+  private final HistogramExemplarSampler origHistogramExemplarSampler = ExemplarConfig.getHistogramExemplarSampler();
+
+  @Before
+  public void setUp() {
+    registry = new CollectorRegistry();
+    TestExemplarSampler defaultExemplarSampler = new TestExemplarSampler(defaultTraceId, defaultSpanId, timestamp);
+    ExemplarConfig.setCounterExemplarSampler(defaultExemplarSampler);
+    ExemplarConfig.setHistogramExemplarSampler(defaultExemplarSampler);
+  }
+
+  @After
+  public void tearDown() {
+    ExemplarConfig.setCounterExemplarSampler(origCounterExemplarSampler);
+    ExemplarConfig.setHistogramExemplarSampler(origHistogramExemplarSampler);
+  }
+
+  @Test
+  public void testCounterNoLabelsDefaultExemplar() throws IOException {
+    Counter noLabelsDefaultExemplar = Counter.build()
+        .name("no_labels_default_exemplar")
+        .help("help")
+        .register(registry);
+    noLabelsDefaultExemplar.inc(3);
+    assertOpenMetrics100Format("no_labels_default_exemplar_total 3.0 # " + defaultExemplarLabels + " 3.0 " + timestampString() + "\n");
+    assert004Format("no_labels_default_exemplar_total 3.0\n");
+    noLabelsDefaultExemplar.inc(2);
+    // The TestExemplarSampler always produces a new Exemplar.
+    // The Exemplar with value 3.0 should be replaced with an Exemplar with value 5.0.
```

Comment:
This comment isn't right, the exemplar value is 2. It's the counter value that is 5.

Reply:
fixed